index.html

```diff
@@ -1603,6 +1604,320 @@ <h2>
     </ul>
 </section>
 
+<section>
+  <h2>
+    Automation
+  </h2>
+  <p>
+    The Compute Pressure API pose a challenge to test authors, as fully
+    exercising interface requires physical hardware devices that respond in predictable ways.
+
+    To address this challenge this document defines a [[WEBDRIVER2]] [=extension
+    commands=] that allows defining and controlling virtual pressure sources that behave
+    like real ones and which can have particular properties and whose readings can be
+    entirely defined by users.
+  </p>
+  <h2>
+    Virtual Pressure Source
+  </h2>
+  <p>
+    A <dfn>virtual pressure source</dfn> simulates the behavior of a real pressure source in controlled ways.
+    It reports pressure changes to zero or more [=platform collectors=] connected to it.
+  </p>
+  <p>
+    Each [=virtual pressure source=] has:
+    <ul>
+      <li>
+        a <dfn>pending state updates</dfn> [=queue=] of zero or more [=pressure states=], which is initially empty.
+      </li>
+      <li>
+        a <dfn>sample update interval</dfn> positive number, initially set to 0 (update immidiately).
+      </li>
+    </ul>
+  </p>
+  <p>
+    A [=virtual pressure source=] is a string that represents a pressure source of a given [=source type=].
+  </p>
+  <p>
+    Each [=top-level traversable=] has a <dfn>virtual pressure source mapping</dfn>, which is an [=ordered map=] of
+    [=source types=] to [=virtual pressure source=].
+  </p>
+  <p>
+    A <dfn>mock platform collector</dfn> is a [=platform collector=] that uses data from the [=virtual pressure sources=]
+    in an [=implementation-defined=] manner.
+  </p>
+  <h3>
+    Extension Commands
+  </h3>
+  <h4>
+    Create virtual pressure source
```

Comment:
In Generic Sensor there was this "connected" parameter (https://www.w3.org/TR/generic-sensor/#create-virtual-sensor-command) which enabled simulate failure (also connected to other chapters in spec e.g. https://www.w3.org/TR/generic-sensor/#sensor-start). In "Update virtual pressure source" it says "If sampleInterval is < 0, stop updating the mock platform collector.", but I do not find connection to rest of the text/other chapters in spec or returning "NotReadableError" like in Generic Sensor.

Reply:
If it's below 0 it just doesn't sent updates, it's like paused

Reply:
Juha is referring to something else: there doesn't seem to be a way to force `observe()` to fail with NotSupportedError in the current version.

Reply:
Do we actually need to test "cpu" and have that fail, or can you just test "invalid-source", because that should fail with NotSupportedError.

Reply:
This would fail with a TypeError, since it's not part of the PressureSource enum. The idea is to test that an `observe()` request reaches the backend and fails as if the OS/platform did not support the requested type.

Reply:
I've implemented this in afcabae581f478015b8125095be8aab67772ed9d